test/shared/src/main/scala/zio/test/DefaultTestReporter.scala

```diff
@@ -77,6 +77,7 @@ object DefaultTestReporter {
       res <- render(executedSpec.mapLabel(_.toString))
       _   <- ZIO.foreach(res.flatMap(_.rendered))(TestLogger.logLine)
       _   <- logStats(duration, executedSpec)
+      _   <- renderTimed(executedSpec).flatMap(_.fold[URIO[TestLogger, Unit]](URIO.unit)(TestLogger.logLine))
```

Comment:
I think we are missing some type of interface something like this:

```scala
type TestAnnotationRenderer[A] = 
  TestAnnotation[A] => RenderedResult
```

Then we can push knowledge of which aspects need rendering to a higher level, and get it out of here, so the user can pass their own rendering for custom test aspects (of course, we will supply the ones for the built in test annotations like timing, etc.).

Reply:
Agreed, I was thinking about whether test annotations should contain their own logic for rendering them but was worried that was coupling data with presentation too much. This is a better way to do that. Let me work on adding. Though I think we may need to modify it slightly to be something like

```scala
type TestAnnotationRenderer[A] =
  (TestAnnotation[A], ExecutedSpec[Any, Any, Any]) => String
```

Conceptually I think we want the renderer to be able to introspect the test annotations from all the tests in a spec (e.g. to sort the rendered test times from longest to shortest or to report at the level of suites instead of individual tests).